### PR TITLE
adjust to new MirageOS libs: use mirage-block{-lwt}, error handling, …

### PR DESCRIPTION
**Pull request by hannesm**
…deduplication of type info

this should make travis green

### CODE REVIEW

File: lib/block_ccm.mli [hunk 1]
```diff
@@ -1,9 +1,12 @@
...
-             with type 'a io = 'a Lwt.t
-              and type page_aligned_buffer = Cstruct.t) : sig
-  include Mirage_types.BLOCK
-    with type 'a io = 'a Lwt.t
-     and type page_aligned_buffer = Cstruct.t
-
-  val connect : ?maclen:int -> ?nonce_len:int -> key:Cstruct.t -> B.t -> t io
+module Make (B: Mirage_block_lwt.S) : sig
+
+  type error = private [> Mirage_device.error | `DecryptError ]
```
Comment (samoht) [inline lib/block_ccm.mli:3]: you don't need `>` here, you can expose the exact type if you want.

Comment (hannesm) [reply lib/block_ccm.mli:3]: as in https://github.com/sg2342/mirage-block-ccm/pull/22 ?

Comment (hannesm) [reply lib/block_ccm.mli:3]: (but I'm still not sure whether this is correct... it get's (as argument) a `B`, and shouldn't the error contain `B.error | [whatever]`?  (but then, I get that error that B.error is not a variant type)...

Comment (samoht) [reply lib/block_ccm.mli:3]: If you want the clients to known exactly all the possible errors, you can just write:
```
type error = [ Mirage_device.error | `DecryptError ]
```

**Pull request merged**